subprojects/base-services/src/main/java/org/gradle/internal/exceptions/NonGradleCauseException.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.exceptions;
+
+/**
+ * An exception that is not caused by a Gradle specific problem.
+ * e.g. a problem caused by a compilation error.
+ * This serves as a marker interface.
```

Comment:
💭 A little off-topic, we have quite a few interfaces that claim in their javadoc to be marker interfaces.

This one is simple and obviously fine.  But running a quick search reveals "marker" interfaces subtyping interfaces with methods, containing inner types, unioning multiple extended interfaces, or even declaring methods!  The fact that we do these things makes me think we aren't working with a very firm definition of what we mean by marker interface.

Have we ever thought about adding some sort of `@Marker` annotation to them, to verify via an annotation processor that they actually are (by whatever criteria we use) and _remain_ marker interfaces?

This might be a good topic for the Architecture Council.

Reply:
I added an [issue](https://github.com/orgs/gradle/projects/59?pane=issue&itemId=23609787).